evolver/hardware/interface.py

```diff
@@ -1,56 +1,52 @@
-import pydantic
-from abc import ABC, abstractmethod
+from abc import abstractmethod
 
+from evolver.base import BaseConfig, BaseInterface
 
-class VialConfigBaseModel(pydantic.BaseModel):
+
+class VialConfigBaseModel(BaseConfig):
     vials: list[int] | None = None
 
 
-class VialBaseModel(pydantic.BaseModel):
+class VialBaseModel(BaseConfig):
     vial: int
 
 
-class BaseCalibrator(ABC):
-    class Config(pydantic.BaseModel):
+class BaseCalibrator(BaseInterface):
+    class Config(BaseConfig):
         calibfile: str = None
 
-    def __init__(self, evovler = None, config: Config = Config()):
-        self.config = config
+    def __init__(self, *args, evovler=None, **kwargs):
+        self.evovler = evovler
+        super().__init__(*args, **kwargs)
 
     @property
     @abstractmethod
     def status(self):
         pass
 
 
-class HardwareDriver(ABC):
-    class Config(pydantic.BaseModel):
+class HardwareDriver(BaseInterface):
+    class Config(BaseConfig):
         pass
     calibrator = None
 
-    def __init__(self, evolver, config = None, calibrator = None):
+    def __init__(self, *args, evolver=None, calibrator=None, **kwargs):
         self.evolver = evolver
-        self.reconfigure(config or self.Config())
         if calibrator:
             self.calibrator = calibrator
-
-    def reconfigure(self, config):
-        self.config = config
+        super().__init__(*args, **kwargs)
 
 
 class VialHardwareDriver(HardwareDriver):
-    def reconfigure(self, config):
-        super().reconfigure(config)
-        if config.vials is None:
-            config.vials = self.evolver.config.vials
-        else:
-            if not set(config.vials).issubset(self.evolver.all_vials):
-                raise ValueError('invalid vials found in config')
+    def __init__(self, *args, vials=None, **kwargs):
+        self.vials = vials if vials else list(range(16))
```

Comment:
I've changed this to ``list(range(settings.DEFAULT_NUMBER_OF_VIALS_PER_BOX))``.

We still need to conclude our discussion about how and to what extent (if at all) centrally involved the evolver manager is. I don't think it would be wise for us to entangle it to the degree that it gets used for dynamic default values.